pkg/downloader/chart_downloader.go

```diff
@@ -94,12 +95,31 @@ func (c *ChartDownloader) DownloadTo(ref, version, dest string) (string, *proven
 		return "", nil, err
 	}
 
-	data, err := g.Get(u.String(), c.Options...)
+	downloadURL := u.String()
+	name := filepath.Base(u.Path)
+
+	if _, ok := g.(*registry.Getter); ok {
+		parts := strings.Split(filepath.Base(u.Path), ":")
+
+		if len(parts) == 1 && version == "" {
+			return "", nil, errors.New("no version or tag provided")
```

Comment:
So, let me get this straight... The `repository` field is required to pass a tag in the URL. Right?

Doesn't this go against the discussion in https://github.com/helm/helm/issues/8332#issuecomment-648287823?

Reply:
That is valid input. We only error if **neither** version or tag is provided. 

`if len(parts) == 1` => no tag specified `&& version == ""` => no version specified